docs/event/assets.md

```diff
@@ -0,0 +1,101 @@
+
+
+### Streams
+
+A stream has the following properties:
+- `language: LangCode?`
+- `type: "video" | "audio" | "text"`: type of the track. Depending on the type, there are additional properties:
+  - `"video"`
+    - `resolution: [uint32, uint32]`
+    - `framerate: "dynamic" | float32`<sup>?</sup>
+    - `codec: "H264" | "H265" | "VP8" | "VP9" | "AV1" | ...`<sup>?</sup>
+  - `"audio"`
+    - `codec: "AAC" | "MP3" | "Opus" | ...`<sup>?</sup>
+  - `"text"`:
+    - `kind: string`: the kind of data this text track represents.
+      - `"subtitle"`: subtitles (does *not* contain speaker names, sounds, and the like)
+      - `"caption"`: closed captions (*does* contain speaker names, sounds, or the like)
+      - `"chapters"`: chapter markers with title per chapter
+      - `"in-video-text"`<sup>(4?)</sup>: representing text in the video (e.g. slide text)
+    - `origin: "manual" | "generated" | null`: describes how this track was created, `manual` meaning it was human crafted, while `generated` means it was automatically generated.
+    - `generator: string?`: non-empty name of the generating software (should only be set if `origin: "generated"`). Example: `"whisper"`.
+
+## Attachments
+
+Attachments are assets without "time-component", unlike tracks.
+They have the following additional metadata:
+- `language: LangCode?`
+
+There are some built-in attachments that have special meanings and requirements, each identified by the flavor:
+
+- `oc/thumbnail`: Thumbnail for the event, e.g. a preview image.
+  - `mimeType` must be `image/*`
+  - `properties` must include `w` and `h`, both holding numbers describing the width and height of the image.
+- `?/timeline-preview`: a sprite image, holding a raster of smaller images, all extracted from the video at regular intervals.
+  - `mimeType` must be `image/*`
+  - `properties` must include:
+    - `imageCountX`, `imageCountY`: how many smaller images are in each row/column
+    - `imageSizeX`, `imageSizeY`: size of each smaller image in pixels
+  - TODO: in the future, this might be better encoded as actual video file
+- `?/segment-preview`<sup>(6?)</sup>: image that is a preview for a segment of the video.
+  - `mimeType` must be `image/*`
+  - `properties` must include `startTime: Milliseconds`, denoting when the segment starts.
+
+TODO: generally make clear what properties are "user changable" and which are automatically set by OC, derived from files.
+
+---
+
+:::danger[Open questions]
+
+- (1?) Is it OK to require ASCII-only for tags and flavors?
+- (2?) Do we really need the `updated` field?
+- (3?) What permissions should be required to read internal assets? Is `write` access to the event enough? And/or should a special role be required?
+- (4?) Better name for this? `on-screen-text`? `text`? `video-text`?
+- (5?) Is `isLive` per track really the correct model? Should this be attached to the event instead? Like, how would a Tobira or LMS decide whether to display an event as live or not?
```

Comment:
In general, video systems backed by Opencast should use flavors / tags to select elements to play.